8.frontend-react/managerapp/src/Components/CRUD/CRUD.component.jsx

```diff
@@ -0,0 +1,214 @@
+  const [loading,setLoading]=useState(true);
+  const [open, setOpen] = useState(false);
+  const [actionType, setActionType] = useState("");
+  const [userdata, setUsersData] = useState([]);
+  const [row, setRow] = useState("");
+    const newColumn=  {
+        Header: "",
+        accessor: "id",
+        Cell: ({ row }) => (
+          <div className="actionsContainer">
+            <RiDeleteBin5Fill
+              className="actionButton"
+              onClick={() => handleRemoveOption(row)}
+            />
+            <MdModeEdit
+              className="actionButton"
+              onClick={() => handleModify(row)}
+            />
+          </div>
+        ), // accessor is the "key" in the data
+      }
+    let newArr=props.columnArr.map((x) => x);
+    newArr.push(newColumn);
+
+    
+  const columns = React.useMemo(
+    
+    () => newArr,
+    []
+  );
+
+  const handleModify = (row) => {
+    setInputValueStr(row.original);
+    console.log(inputValueStr)
+    setActionType("modify");
+    setRow(row);
+    setOpen(true);
+  };
+
+  const handleAdd=()=>{
+    setActionType("add");
+    setOpen(true);
+  }
+
+
+  const handleRemoveOption = (row) => {
+    setActionType("remove");
+    setRow(row);
+    setOpen(true);
+
+  };
+
+
+  const updateTable = () => {
+    setLoading(true);
+    let object = "";
+    let objectData="";
+    props.fetchData().then(object=>{
+        for (var prop in object.data) {
+            objectData=object.data[prop];
+            break;
+        }
+        setUsersData(objectData);
+    })
+    .catch(err=>{
+        console.log(err);
+    })
+
+    setTimeout(() => {
+      setLoading(false);
+  }, 1000);
+  };
+
+  const handleOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+const remove=()=>{
+    setLoading(true);
+    props.confirmRemove(row.original.id).then(()=>{
+        updateTable();
+        handleClose();
+    }).catch(err=>{
+        console.log(err);
+    })
+}
+  const add= (e)=>{
+      setLoading(true);
+    props.confirmAdd(e).then(()=>{
+        updateTable();
+        handleClose();
+    }).catch(err=>{
+        console.log(err);
+    })
+  }
+
+
+  const update=(e)=>{
+     
+    setLoading(true);
+      props.confirmUpdate(e,row.original.id).then(()=>{
+        updateTable();
+        handleClose();
+
+      })
+      .catch(err=>{
+          console.log(err);
+      });
+     ;  }
+
+  useEffect(() => {
+  
+    updateTable();
+  //   setTimeout(() => {
+  //     setLoading(false);
+  // }, 1500);
+  }, []);
+
+  const customStyles = {
+    content: {
+      top: "50%",
+      left: "50%",
+      right: "auto",
+      bottom: "auto",
+      marginRight: "-50%",
+      transform: "translate(-50%, -50%)",
+      display: "flex",
+      "flex-direction": "column",
+      "align-items": "center",
+    },
+  };
+
+  if(loading){
+    return(
+    
+    <Loading/>
+    );
+  }
+  return (
+    <div style={{marginLeft:'25vw'}}>
```

Comment:
please use the following : 

import styles from "../example.css"

usage:

<div className={styles.nadav}>